app/components/graph-editor/drawable-interfaces.ts

```diff
@@ -61,6 +55,24 @@ export interface DrawableGraph {
     readonly edges: Iterable<DrawableEdge>;
 
     /**
+     * selectedItems  
+     *   The set of selected graph elements.
+     */
+    readonly selectedItems: Set<Drawable>;
+
+    /**
+     * origin  
+     *   The displacement of the origin point of the graph editor canvas.
+     */
+    origin: { x: number, y: number };
```

Comment:
I'm not sure I like the idea of exporting the `point` type from `canvas.ts`, but I see your... argument.